api/types/filters/parse.go

```diff
@@ -196,6 +195,37 @@ func (args Args) Match(field, source string) bool {
 	return false
 }
 
+// GetBoolOrDefault returns a boolean value of the key if the key is present
+// and is intepretable as a boolean value. Otherwise the default value is returned.
+// Error is not nil only if the filter values are not valid boolean or are conflicting.
+func (args Args) GetBoolOrDefault(key string, defaultValue bool) (bool, error) {
+	fieldValues, ok := args.fields[key]
+
+	if !ok {
+		return defaultValue, nil
+	}
+
+	if len(fieldValues) == 0 {
+		return defaultValue, invalidFilter{key, nil}
+	}
+
+	isFalse := fieldValues["0"] || fieldValues["false"]
+	isTrue := fieldValues["1"] || fieldValues["true"]
+
+	conflicting := isFalse && isTrue
+	invalid := !isFalse && !isTrue
+
+	if conflicting || invalid {
```

Comment:
This could all just be `if isFalse == isTrue {`, right?

Reply:
Hmm, don't we also want to error out if we have any values that aren't one of the four supported strings? (or is that a breaking change from the previous implementation?)

Reply:
Yes, that's the `invalid` check (true if value is none of [0,1,true,false]) and it ends with the error.
There are tests to prove that 😄  :
https://github.com/moby/moby/blob/18e30a04ec5e7d332bd9e81e35701de03f5382b8/api/types/filters/parse_test.go#L451-L468

Reply:
Yes, this is logically the same as `if isFalse == isTrue`, but it looked confusing (looks like false == true).
I went with `if conflicting || invalid` instead as it felt clearer and explicit about when this branch is taken (when the filter values are conflicting, or are invalid).
But maybe that's just my perception 😅